app/src/app/components/users-list/users-list.component.ts

```diff
@@ -0,0 +1,235 @@
+import { Component, OnInit, ViewChild, ViewEncapsulation, Input, EventEmitter, Output } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
+import { UserAccount } from 'src/app/classes/authn/user-account';
+import { Paginated, RestApiConnectorService } from 'src/app/services/connectors/rest-api/rest-api-connector.service';
+import { Role } from 'src/app/classes/authn/role';
+import { MatPaginator } from '@angular/material/paginator';
+import { AuthenticationService } from '../../services/connectors/authentication/authentication.service';
+import { Status } from 'src/app/classes/authn/status';
+// import { Team } from 'src/app/classes/authn/team';
+import { SelectionModel } from '@angular/cdk/collections';
+
+@Component({
+  selector: 'app-users-list',
+  templateUrl: './users-list.component.html',
+  styleUrls: ['./users-list.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class UsersListComponent implements OnInit {
+  @Input() config:UsersListConfig;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @Output() public onSelect = new EventEmitter<String>();
+  public userAccounts$: Observable<Paginated<UserAccount>>;
+  public userAccounts: UserAccount[];
+  public columnsToDisplay: string[];
+  public filterOptions: any[];
+  public totalObjectCount: number = 0;
+  public userSubscription: Subscription;
+  public selectedFilters: string[];
+  public searchQuery: '';
+  // public team:Team;
+  public team:any;
+  public selection: SelectionModel<String>;
+
+  /**
+   * Whether or not to show the search
+   */
+  public get showSearch(): boolean {
+    return this.config && this.config.showSearch !== null && this.config.showSearch !== undefined ? this.config.showSearch : true;
```

Comment:
Since `null` and `undefined` are both falsy values, it isn't necessary to check them separately. This statement can be condensed to `return this.config && this.config.showSearch ? this.config.showSearch: true;`. It also avoids some repetition which will make the code easier to read and maintain.